src/figure2dot.m

```diff
@@ -29,127 +29,108 @@ function figure2dot(filename)
 %   POSSIBILITY OF SUCH DAMAGE.
 %
 % =========================================================================
-  global node_number
-
-  % also show hidden handles
-  set(0, 'ShowHiddenHandles', 'on');
-
-  filehandle = fopen(filename, 'w');
-  finally_fclose_filehandle = onCleanup(@() fclose(filehandle));
-
-  % start printing
-  fprintf(filehandle, 'digraph simple_hierarchy {\n\n');
-
-  fprintf(filehandle, 'node[shape=box];\n\n');
-
-  % define the root node
-  node_number = 0;
-  p = get(gcf, 'Parent');
-  % define root element
-  type = get(p, 'Type');
-  fprintf(filehandle, 'N%d [label="%s"]\n\n', node_number, type);
-
-  % start recursion
-  plot_children(filehandle, p, node_number);
-
-  % finish off
-  fprintf(filehandle, '}');
-  set(0, 'ShowHiddenHandles', 'off');
 
+    ipp = m2tInputParser();
+    ipp = ipp.addRequired(ipp, 'filename', @ischar);
+    ipp = ipp.addParamValue(ipp, 'object', gcf, @ishghandle);
+    ipp = ipp.parse(ipp, filename, varargin{:});
+    args = ipp.Results;
+    %TODO: improve documentation of parameters of figure2dot
+
+    filehandle = fopen(args.filename, 'w');
+    finally_fclose_filehandle = onCleanup(@() fclose(filehandle));
+
+    % start printing
+    fprintf(filehandle, 'digraph simple_hierarchy {\n\n');
+    fprintf(filehandle, 'node[shape=box];\n\n');
+
+    % define the root node
+    node_number = 0;
+    p = get(args.object, 'Parent');
+    % define root element
+    type = get(p, 'Type');
+    fprintf(filehandle, 'N%d [label="%s"]\n\n', node_number, type);
+
+    % start recursion
+    plot_children(filehandle, p, node_number);
+
+    % finish off
+    fprintf(filehandle, '}');
+
+    % ----------------------------------------------------------------------------
+    function plot_children(fh, h, parent_node)
+
+        children = allchild(h);
+
+        for h = children(:)'
+            if shouldSkip(h), continue, end;
+            node_number = node_number + 1;
+
+            label = {};
+            label = addHGProperty(label, h, 'Type', '');
+            try
+                hClass = class(handle(child));
```

Comment:
This will always fail since `child` is not defined anymore, but should be `h`. For Octave you could retrieve the 'type' property instead.